File: backend/internal/dto/operational/tracker_reminder.go
package operational

type UpdateTrackerReminderConfigRequest struct {
	Enabled               bool   `json:"enabled"`
	StartHour             int    `json:"start_hour" validate:"gte=0,lte=23"`
	EndHour               int    `json:"end_hour" validate:"gte=1,lte=24"`
	WeekdaysOnly          bool   `json:"weekdays_only"`
	Timezone              string `json:"timezone" validate:"required,max=64"`
	HeartbeatStaleMinutes int    `json:"heartbeat_stale_minutes" validate:"gte=1,lte=1440"`
	NotifyInApp           bool   `json:"notify_in_app"`
	NotifyWhatsapp        bool   `json:"notify_whatsapp"`
}

type TrackerReminderConfigResponse struct {
	TenantID              string  `json:"tenant_id"`
	Enabled               bool    `json:"enabled"`
	StartHour             int     `json:"start_hour"`
	EndHour               int     `json:"end_hour"`
	WeekdaysOnly          bool    `json:"weekdays_only"`
	Timezone              string  `json:"timezone"`
	HeartbeatStaleMinutes int     `json:"heartbeat_stale_minutes"`
	NotifyInApp           bool    `json:"notify_in_app"`
	NotifyWhatsapp        bool    `json:"notify_whatsapp"`
	NextReminderAt        *string `json:"next_reminder_at,omitempty"`
	UpdatedAt             string  `json:"updated_at"`
}

type TrackerReminderTestResponse struct {
	DeliveredInApp    bool    `json:"delivered_in_app"`
	DeliveredWhatsapp bool    `json:"delivered_whatsapp"`
	WhatsappError     *string `json:"whatsapp_error,omitempty"`
}

File: frontend/src/services/tracker-reminder.ts
import { authGetJSON, authPostJSON, authRequestJSON } from "@/lib/api-client";

export interface TrackerReminderConfig {
  tenant_id: string;
  enabled: boolean;
  start_hour: number;
  end_hour: number;
  weekdays_only: boolean;
  timezone: string;
  heartbeat_stale_minutes: number;
  notify_in_app: boolean;
  notify_whatsapp: boolean;
  next_reminder_at?: string | null;
  updated_at: string;
}

export interface UpdateTrackerReminderConfigPayload {
  enabled: boolean;
  start_hour: number;
  end_hour: number;
  weekdays_only: boolean;
  timezone: string;
  heartbeat_stale_minutes: number;
  notify_in_app: boolean;
  notify_whatsapp: boolean;
}

export interface TrackerReminderTestResult {
  delivered_in_app: boolean;
  delivered_whatsapp: boolean;
  whatsapp_error?: string | null;
}

export const trackerReminderKeys = {
  config: () => ["operational", "tracker", "reminder-config"] as const,
};

export async function getTrackerReminderConfig() {
  return authGetJSON<TrackerReminderConfig>("/tracker/reminder-config");
}

export async function updateTrackerReminderConfig(payload: UpdateTrackerReminderConfigPayload) {
  return authRequestJSON<TrackerReminderConfig>("/tracker/reminder-config", {
    method: "PUT",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  });
}

export async function sendTrackerReminderTest() {
  return authPostJSON<TrackerReminderTestResult, Record<string, never>>(
    "/tracker/reminder-config/test",
    {},
  );
}
